### PR TITLE
simple url Unit Test

### PR DESCRIPTION
**Pull request by RomanGavrilovM**
add testing for Null and Equals in Rapid Response

### CODE REVIEW

File: app/src/main/java/com/example/nasaapod/ui/apod/NasaApodPageFragment.kt [hunk 1]
```diff
@@ -100,32 +97,40 @@ class NasaApodPageFragment : Fragment(R.layout.fragment_page_apod) {
                 0 -> {
                     viewLifecycleOwner.lifecycle.coroutineScope.launchWhenStarted {
                         apodViewModel.today.collect() { day ->
-                            day?.let {
-                                binding.apodImg.load(day.url)
+                            day.let {
+                                if (validator.isValidApodUrl(day.url)) {
```
Comment (AlexShutov) [inline app/src/main/java/com/example/nasaapod/ui/apod/NasaApodPageFragment.kt:101]: логику в презентер

File: app/src/main/java/com/example/nasaapod/ui/apod/NasaApodPageFragment.kt [hunk 2]
```diff
@@ -180,6 +185,11 @@ class NasaApodPageFragment : Fragment(R.layout.fragment_page_apod) {
         return (dp / Resources.getSystem().displayMetrics.density).toInt()
     }
 
+    fun validateUrl(url: String): Boolean {
+        return url.contains(".jpg")
+
+    }
+
```
Comment (AlexShutov) [inline app/src/main/java/com/example/nasaapod/ui/apod/NasaApodPageFragment.kt:192]: кодстайл- не нужны пустые строки

File: app/src/main/java/com/example/nasaapod/ui/apod/NasaApodPageFragment.kt (hunk 1 above)
Comment (AlexShutov) [reply app/src/main/java/com/example/nasaapod/ui/apod/NasaApodPageFragment.kt:101]: (во вьюмодель)

File: app/src/main/java/com/example/nasaapod/ui/apod/ApodResponseValidator.kt [hunk 3]
```diff
@@ -0,0 +1,35 @@
+package com.example.nasaapod.ui.apod
+
+import com.example.nasaapod.api.apod.ApodResponse
+import java.util.regex.Pattern
+
+class ApodResponseValidator() {
```
Comment (AlexShutov) [inline app/src/main/java/com/example/nasaapod/ui/apod/ApodResponseValidator.kt:6]: а где сам тест?

File: app/src/test/java/com/example/nasaapod/ui/apod/ApodResponseValidatorTest.kt [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    }
+
+    @Test
+    fun nasaApodUrlValidator_NullUrl_ReturnFalse() {
+        assertNotNull(ApodResponseValidator.isValidApodUrl(null))
+    }
+
+    //это видим о будут моки
+    val mockResponseToday = ApodResponse(
+        "today date",
```
Comment (AlexShutov) [inline app/src/test/java/com/example/nasaapod/ui/apod/ApodResponseValidatorTest.kt:36]: датаклассы не мокаются, если датакласс- просто создавать и сравнивать через assertEquals

File: app/src/test/java/com/example/nasaapod/ui/apod/ApodResponseValidatorTest.kt [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    }
+
+    @Test
+    fun nasaApodResponseValidator_SimpleFalse(){
+        assertEquals(mockResponseToday,mockResponseDayAgo)
+    }
+
+    @Test
+    fun nasaApodResponseValidator_EqualsTrue(){
+        assertEquals(mockResponseToday,mockResponseTodayDublicate)
```
Comment (AlexShutov) [inline app/src/test/java/com/example/nasaapod/ui/apod/ApodResponseValidatorTest.kt:78]: нет смысла сравнивать 2 заранее созданных в тесте обьекта, нужно ожидаемый и тот, что вернулся от вызова тестируемого метода

File: app/src/test/java/com/example/nasaapod/ui/apod/ApodResponseValidatorTest.kt [hunk 6]
```diff
@@ -0,0 +1,81 @@
+package com.example.nasaapod.ui.apod
+
+import com.example.nasaapod.api.apod.ApodResponse
```
Comment (AlexShutov) [inline app/src/test/java/com/example/nasaapod/ui/apod/ApodResponseValidatorTest.kt:3]: тест лучше бы отдельным коммитом, чтобы не было большого списка изменений
